### PR TITLE
Refactor cli argument parsing, support single quote assignments

### LINKED ISSUE #53: Quoting values in CLI not possible
When passing an argument via CLI that contains whitespace, it would be good if quoting was supported.

```kotlin
data class Configuration(
    val notesFileName: String
)
```

`--notes-file-name "My Clippings.txt"`

### PR DESCRIPTION
**Pull request by AndreasVolkmann**
Closes #53 

Ok found that most quoting is functional.

Only one corner case that I could find is:
Single-quoted cli assignment, i.e. `'argument=expected value'`

### CODE REVIEW

File: arkenv/src/test/kotlin/com/apurebase/arkenv/feature/cli/CliArgumentParserTest.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+internal class CliArgumentParserTest {
+
+    private data class TestCase(val name: String, val input: List<String>, val expected: String)
+
+    private val testCases = listOf(
+        TestCase("no quotes", listOf("hello", "world"), "hello"),
+        TestCase("double quotes", listOf("\"hello", "world\""), "hello world"),
+        TestCase("single quote inside value", listOf("D'vloper"), "D'vloper"),
+        TestCase("single quoted", listOf("'test", "expected'"), "test expected"),
+        TestCase("single quoted, containing singlq quote", listOf("'test", "ex'pected'"), "test ex'pected"),
```
Comment (jeggy) [inline arkenv/src/test/kotlin/com/apurebase/arkenv/feature/cli/CliArgumentParserTest.kt:21]: Maybe I'm thinking wrong. But shouldn't a case like this work also?
`TestCase("single quoted, containing singlq quote", listOf("'test", "ex\\'pected'"), "test ex'pected"),`

Comment (AndreasVolkmann) [reply arkenv/src/test/kotlin/com/apurebase/arkenv/feature/cli/CliArgumentParserTest.kt:21]: It works without the backslash already, why would you need to provide the backslash?

Comment (AndreasVolkmann) [reply arkenv/src/test/kotlin/com/apurebase/arkenv/feature/cli/CliArgumentParserTest.kt:21]: If you provide `--argument 'test\' expected'`, then it might make sense. 
I.e. if you want a quote at the end of a word (cli is split by space) but the value continues further.

Comment (AndreasVolkmann) [reply arkenv/src/test/kotlin/com/apurebase/arkenv/feature/cli/CliArgumentParserTest.kt:21]: Oh, another case is: `--argument 'test'`
Should the output be `'test'` or `test`?

Comment (jeggy) [reply arkenv/src/test/kotlin/com/apurebase/arkenv/feature/cli/CliArgumentParserTest.kt:21]: I have no clue what the correct solution for this is. When using bash and cli tools I always just try stuff until it works :laughing: 

So maybe we can just leave it as is. Which means anything will work, as `main(args: List<String>)` is already splitt by spaces and you would just look at the first and last characters to merge them together.

**Review by jeggy — APPROVED** [review]
